Fixes required for use with multiple Cloud Environments

dougalII opened the pull request:
##### SUMMARY
Fixes credential_scopes for track2 authentication when connecting to non "Azure Public" cloud environments
Fixes for AKV to correct resource url for non "Azure Public" cloud environments

fixes: #331 #836 #702 

##### ISSUE TYPE
<!--- Pick one below and delete the rest -->
- Bugfix Pull Request

##### COMPONENT NAME

- azure_rm_common
- azure_rm_keyvaultkey
- azure_rm_keyvaultsecret
- azure_rm_keyvaultkey_info
- azure_rm_keyvaultsecret_info

##### ADDITIONAL INFORMATION
These changes allow you to use other cloud platforms in azure.

The track2 fixes are related to an azure-sdk issue i created: [here](https://github.com/Azure/azure-sdk-for-python/issues/24440) it uses poorly documented keyword credential_scopes to tell the SDK which environment the token request is for.

The AKV Changes are to remove the hard coded ` resource="https://vault.azure.net"` as each cloud environment has its own resource url for AKV.

Commit:
fix(track2_auth): Fixed Track2 Auth to include credential_scope, this is a requirement when used with alternative azure cloud environments.
fix(akv_url): Fixed AKV Resource URL, this is required as different azure cloud environments.

Fred-sun commented on plugins/module_utils/azure_rm_common.py:
```diff
@@ -891,12 +891,12 @@ def get_mgmt_svc_client(self, client_type, base_url=None, api_version=None, supp
         else:
             if is_track2:
-                client_kwargs = dict(credential=self.azure_auth.azure_credential_track2, subscription_id=mgmt_subscription_id, base_url=base_url)
+                client_kwargs = dict(credential=self.azure_auth.azure_credential_track2, subscription_id=mgmt_subscription_id, base_url=base_url, credential_scopes=[base_url + ".default"])
```
line too long (188 > 160 characters)
```suggestion
                client_kwargs = dict(credential=self.azure_auth.azure_credential_track2, subscription_id=mgmt_subscription_id, base_url=base_url, credential_scopes=[base_url + ".default"])
```

dougalII replied:
implemented suggestion, sorry, could not get the "commit suggestion" to actually work.

Fred-sun commented on plugins/modules/azure_rm_keyvaultkey.py:
```diff
@@ -176,20 +176,21 @@ def exec_module(self, **kwargs):
         if self.module.params['auth_source'] == 'msi':
             try:
                 self.log("Get KeyVaultClient from MSI")
                 resource = self.azure_auth._cloud_environment.suffixes.keyvault_dns.split('.', 1).pop()
```
never userd
```e.g.
resource = self.azure_auth._cloud_environment.suffixes.keyvault_dns.split('.', 1).pop()

dougalII replied:
removed, thank you

Fred-sun commented on plugins/modules/azure_rm_keyvaultkey_info.py:
```diff
@@ -313,20 +313,21 @@ def exec_module(self, **kwargs):
         if self.module.params['auth_source'] == 'msi':
             try:
                 self.log("Get KeyVaultClient from MSI")
                 resource = self.azure_auth._cloud_environment.suffixes.keyvault_dns.split('.', 1).pop()
```
never used
```e.g.
resource = self.azure_auth._cloud_environment.suffixes.keyvault_dns.split('.', 1).pop()

dougalII replied:
removed, thank you

Commit:
fix(cleanup): Removed un-used resource var assignment
fix(line_length): split track2 client_kwargs to 2 lines to keep below the 160 character limit

perrymckenzie commented:
Agree with @rmawhinnie  - our team needs this soon too.  Looking forward to this merge!

danfinn commented:
Hoping to get this merged soon as well.

ramdaspotale commented:
it will be very much helpful if get merged sooner.

Fred-sun commented on plugins/module_utils/azure_rm_common.py:
```diff
@@ -891,12 +891,13 @@ def get_mgmt_svc_client(self, client_type, base_url=None, api_version=None, supp
         if suppress_subscription_id:
             if is_track2:
-                client_kwargs = dict(credential=self.azure_auth.azure_credential_track2, base_url=base_url)
+                client_kwargs = dict(credential=self.azure_auth.azure_credential_track2, base_url=base_url, credential_scopes=[base_url + ".default"])
```
I'm sorry for replying to you so late. Could you please tell me the function of this parameter（credential_scopes) ? Because I didn't apply this parameter in the Track2 SDK. Thank you very much!

dougalII replied:
Hey @Fred-sun,
It essentially sets the management endpoints the generated auth token is for, if this is not set, it defaults to the AzurePublicCloud url, and you get the bellow error when talking to any of the other azure clouds.
`Message: The access token has been obtained for wrong audience or resource 'https://management.azure.com'. It should exactly match with one of the allowed audiences 'https://management.core.usgovcloudapi.net/','https://management.core.usgovcloudapi.net','https://management.usgovcloudapi.net/','https://management.usgovcloudapi.net'.
`

this error was discussed here:
https://github.com/Azure/azure-sdk-for-python/issues/24440

dougalII replied:
@Fred-sun, do you need any more information on this? We are finding we are jumping threw a lot of hoops to get these fixes inside our AWX, EE and Operator-SDK environments.

Fred-sun commented:
@dougalII I will move forward with the merger as soon as possible. Thank you very much!

xuzhang3 commented:
@dougalII LGTM
